base/ltfssdcl.dtx

```diff
@@ -507,8 +569,138 @@
  \expandafter#1\ifx\math@bgroup\bgroup{#4}\else#4\fi
  }%
 }
+%    \end{macrocode}
+%
+%
+
+%  \begin{macro}{\freeze@math@version}
+% \changes{v3.0x}{2021/09/26}{Macro added for (gh/676)}
+%    This command stores the current state of the math version and sets
+%    things up to return to it after each formula from now on. We use
+%    L3 programming layer code to set it up.
+%    \begin{macrocode}
+\ExplSyntaxOn
+\cs_new:Npn\freeze@math@version #1 {
```

Comment:
Maybe `\cs_new_protected:Npn`, to follow guidelines

Reply:
I don't think  this can ever need protection, can you?  So technically I don't think it matters, though it doesn't hurt either I guess. Do we know if `\protect` introduces as (tiny) time penalty or not?

Reply:
There is a (very small) time penalty indeed. My benchmarks indicate typically around 6% slow down for a protected `\prg_do_nothing:` in pdfTeX on Linux.

protected: 2.6e-8 seconds (0.0829 ops)
normal: 2.59e-8 seconds (0.0765 ops)

Reply:
If instead of `\prg_do_nothing:` we benchmark a single assignment `\def\mytmp{}` stored inside a protected and unprotected macro the difference becomes negligible:

protected:
9.47e-8 seconds (0.3 ops)
9.54e-8 seconds (0.305 ops)
9.48e-8 seconds (0.302 ops)

unprotected:
9.32e-8 seconds (0.298 ops)
9.47e-8 seconds (0.3 ops)
9.51e-8 seconds (0.302 ops)

Reply:
interesting here is the execution of a protected macro versus the execution of it unprotected, but in this particular case it doesn't really matter since that command is called at most once per mathversion in a document so ... :-) and not once per formula

Reply:
I tend to do calculations on the ops, not the time. The time is in my experience far more inconsistent than the ops measured by `l3benchmark`. And I got the 6% pretty consistently for a macro that just expands to nothing.

Reply:
Oh, the ops...  I don't know exactly what counts as an “operation” here, but I can see why that can be more stable than the raw time.  But in any case, the extra time it takes for `\protected` _vs._ otherwise makes practically zero difference, so I still favour following `l3styleguide`:

> The expandability of each function should be well-defined. Functions which cannot be fully expanded must be `protected`.

Reply:
This is not at all the best place to discuss this, but )for a few reasons) I have never been happy with this sentence:
"Functions which cannot be fully expanded must be protected."

I am not sure that it is sufficiently precise; and it is definitely an over-simplified requirement.